### DISCUSSION TITLE
System.Net.Http.HttpRequestException when trying to send a Get request to "https://api.telegram.org"

### DISCUSSION THREAD

**Discussion by homostultus39** [nanoframework/Home#1411] [Q&A] [answered]
```
if (result.ConnectionStatus == WifiConnectionStatus.Success)
{
    Debug.WriteLine("Connected to Wifi network");
    string apiUrl = $"https://api.telegram.org/bot{BotToken}/sendMessage?chat_id={ChatId}&text={"Hello from your bot!"}";
    using (HttpClient client = new HttpClient())
    {
        HttpResponseMessage response = client.Get(apiUrl);
    }
}
```
Good day! I encountered this problem, I connect the microcontroller to the network and try to send a get request to the specified address, but the System.Net.Http.HttpRequestException error occurs.
Full error log:
++++ Exception System.Net.Sockets.SocketException - CLR_E_FAIL (3) ++++
     ++++ Message:
     ++++ System.Net.Security.SslNative::SecureConnect [IP: 0000] ++++
     ++++ System.Net.Security.SslStream::Authenticate [IP: 0054] ++++
     ++++ System.Net.Security.SslStream::AuthenticateAsClient [IP: 000a] ++++
     ++++ System.Net.HttpWebRequest::EstablishConnection [IP: 0285] ++++
     ++++ System.Net.HttpWebRequest::SubmitRequest [IP: 001a] ++++
     ++++ System.Net.HttpWebRequest::GetResponse [IP: 000c] ++++
     ++++ System.Net.Http.HttpClientHandler::Send [IP: 00a7] ++++
     ++++ System.Net.Http.HttpMessageInvoker::Send [IP: 0013] ++++
     ++++ System.Net.Http.HttpClient::Send [IP: 006f] ++++
     ++++ System.Net.Http.HttpClient::Get [IP: 000c] ++++
     ++++ sssss.Program::Wifi_AvailableNetworksChanged [IP: 0071] ++++
     ++++ System.Device.Wifi.WifiAdapter::OnAvailableNetworksChangedInternal [IP: 0028] ++++
     ++++ System.Device.Wifi.WifiEventListener::OnEvent [IP: 0022] ++++
     ++++ nanoFramework.Runtime.Events.EventSink::ProcessEvent [IP: 0023] ++++
     ++++ nanoFramework.Runtime.Events.EventSink::EventDispatchCallback [IP: 0014] ++++
     ++++ Exception System.Net.WebException - 0x00000000 (3) ++++
     ++++ Message: GetResponse() failed
     ++++ System.Net.HttpWebRequest::GetResponse [IP: 00da] ++++
     ++++ System.Net.Http.HttpClientHandler::Send [IP: 00a7] ++++
     ++++ System.Net.Http.HttpMessageInvoker::Send [IP: 0013] ++++
     ++++ System.Net.Http.HttpClient::Send [IP: 006f] ++++
     ++++ System.Net.Http.HttpClient::Get [IP: 000c] ++++
     ++++ sssss.Program::Wifi_AvailableNetworksChanged [IP: 0071] ++++
     ++++ System.Device.Wifi.WifiAdapter::OnAvailableNetworksChangedInternal [IP: 0028] ++++
     ++++ System.Device.Wifi.WifiEventListener::OnEvent [IP: 0022] ++++
     ++++ nanoFramework.Runtime.Events.EventSink::ProcessEvent [IP: 0023] ++++
     ++++ nanoFramework.Runtime.Events.EventSink::EventDispatchCallback [IP: 0014] ++++
     ++++ Exception System.Net.Http.HttpRequestException - 0x00000000 (3) ++++
     ++++ Message: An error occurred while sending the request
     ++++ System.Net.Http.HttpClientHandler::Send [IP: 00bf] ++++
     ++++ System.Net.Http.HttpMessageInvoker::Send [IP: 0013] ++++
     ++++ System.Net.Http.HttpClient::Send [IP: 006f] ++++
     ++++ System.Net.Http.HttpClient::Get [IP: 000c] ++++
     ++++ sssss.Program::Wifi_AvailableNetworksChanged [IP: 0071] ++++
     ++++ System.Device.Wifi.WifiAdapter::OnAvailableNetworksChangedInternal [IP: 0028] ++++
     ++++ System.Device.Wifi.WifiEventListener::OnEvent [IP: 0022] ++++
     ++++ nanoFramework.Runtime.Events.EventSink::ProcessEvent [IP: 0023] ++++
     ++++ nanoFramework.Runtime.Events.EventSink::EventDispatchCallback [IP: 0014] ++++

The error occurs in the line
         HttpResponseMessage response = client.Get(apiUrl);

Maybe I want to do something that can't be done, or maybe I'm just doing something wrong, I don't know... Has anyone encountered a similar problem?

**Comment by Ellerbach** [comment, marked as answer]
@homostultus39 which certificate are you using? If none, you'll get an exception!
Keep in mind, there is no certificate store like in Windows, Linux and other OS. You have to specify a certificate for the verification.

Run the following command to see the cert to use:

```shell
openssl s_client -connect api.telegram.org:443 -showcerts
```

**Reply by homostultus39** [reply]
Everything worked out! Thanks a lot!
